orm-rules-tests/rules-test/rules/test_match_method.yml

```diff
@@ -0,0 +1,224 @@
+---
+
+schema_version: 2
+
+rules:
+  - description: Test match method begins_with
+    domains:
+      - test-method
+    matches:
+      all:
+        - paths:
+            exact:
+              - '/method/begins_with'
+        - method:
+            begins_with:
+              - 'G'
```

Comment:
Why allow complex things like this? Do not the RFC specify a short list of allowed methods? Even if that's not the case, when do you ever need to match anything other than one or more full method names?

Reply:
There is a list of standardized methods (https://www.iana.org/assignments/http-methods/http-methods.xhtml), but I can't see that the RFC (https://tools.ietf.org/html/rfc7231#section-4) forbids non-standard methods.

I don't know of any example systems, but the RFC states:
```
   HTTP was originally designed to be usable as an interface to
   distributed object systems.  The request method was envisioned as
   applying semantics to a target resource in much the same way as
   invoking a defined method on an identified object would apply
   semantics.  The method token is case-sensitive because it might be
   used as a gateway to object-based systems with case-sensitive method
   names.
```

The short answer is that I see no reason to restrict the matching specifically for methods, since the RFC allows complex method names.

Another soft reason is that, this way, the rule format looks the same for all string matching making it easier for users to recognize and write matches of different types, and also making the logic simpler in the ORM codebase.

Reply:
> Another soft reason is that, this way, the rule format looks the same for all string matching types making it easier for users to recognize and write matches of different types, and also making the logic simpler in the ORM codebase.

I would call this a good enough reason, unless there are restrictions on methods in the software that we are generating config for. Valid rules should not generate invalid config.